Dockerfile

```diff
@@ -1,46 +1,39 @@
-FROM python:3.5.3-alpine
-MAINTAINER paul.traylor@linecorp.com
+FROM python:3.5.5-alpine
+LABEL maintainer=paul.traylor@linecorp.com
 
-ENV PROMETHEUS_VERSION 2.1.0
-ENV PROMETHEUS_DOWNLOAD_URL https://github.com/prometheus/prometheus/releases/download/v${PROMETHEUS_VERSION}/prometheus-${PROMETHEUS_VERSION}.linux-amd64.tar.gz
-
-RUN adduser -D -u 1000 promgen promgen
-RUN apk add --no-cache --update mariadb-dev build-base bash && \
-    rm -rf /var/cache/apk/*
-
-RUN set -ex; \
-        apk add --no-cache curl tar; \
-        curl -L -s $PROMETHEUS_DOWNLOAD_URL \
-		| tar -xz -C /usr/local/bin --strip-components=1 prometheus-${PROMETHEUS_VERSION}.linux-amd64/promtool; \
-        apk del curl tar; \
-	rm -rf /var/cache/apk;
-
-ENV PYTHONUNBUFFERED 1
-ENV PIP_NO_CACHE_DIR off
-
-RUN mkdir -p /etc/prometheus
-RUN mkdir -p /etc/promgen
-RUN mkdir -p /usr/src/app
-RUN chown promgen /etc/prometheus
+ARG PROMETHEUS_VERSION=2.3.2
+ARG PROMETHEUS_DOWNLOAD_URL=https://github.com/prometheus/prometheus/releases/download/v${PROMETHEUS_VERSION}/prometheus-${PROMETHEUS_VERSION}.linux-amd64.tar.gz
 
 COPY docker/requirements.txt /tmp/requirements.txt
-RUN pip install -r /tmp/requirements.txt
-
 COPY setup.py /usr/src/app/setup.py
 COPY promgen /usr/src/app/promgen
 COPY promgen/tests/examples/promgen.yml /etc/promgen/promgen.yml
+COPY docker/docker-entrypoint.sh /docker-entrypoint.sh
 
 WORKDIR /usr/src/app
-RUN pip install -e .
 
-ENV CONFIG_DIR=/etc/promgen
+ENV PYTHONUNBUFFERED=1 \
+	PROMGEN_CONFIG_DIR=/etc/promgen \
+	STATIC_ROOT=/srv/promgen/www/static
 
-USER promgen
-EXPOSE 8000
+RUN set -ex; \
+	addgroup -g 16395 promgen; \
+	adduser -D -u 16395 -G promgen promgen; \
```

Comment:
Is there something special about the value 16395 here ? Shouldn't be a problem, I'm just curious why you chose this value :)

Reply:
There is nothing special about this value, it is an arbitrary uid >> 1000. This is the random but static uid I wrote about before.

Some projects choose to simply run `adduser` as the first command and hope for the same uid to be allocated every time. For promgen, the uid is of the upmost importance, because if permissions are misconfigured, the worker will be unable to write any configuration files. That's why I think explicitly setting the uid is a good idea.

Because the uid is much greater than 1000, it is unlikely to map to any user on the local system.

Reply:
Ok. Should be fine, I just wish there was more authoritative sources about values like this 😓 
I agree blindly trusting `adduser` is scary.
For my own deployment I use 9090:9090 for Prometheus (to match the default port) but I suppose in Promgen's case we can't easily do something similar.